exam/models/exam.py

```diff
@@ -502,31 +479,70 @@ def onchange_student(self):
     grade_system = fields.Many2one('grade.master', "Grade System",
                                    help="Grade System selected")
     message_ids = fields.One2many('mail.message', 'res_id', 'Messages',
-                                  domain=lambda self: [('model', '=',
-                                                        self._name)],
+                                  domain=lambda self: [
+                                   ('model', '=', self._name)],
                                   auto_join=True)
     message_follower_ids = fields.One2many('mail.followers', 'res_id',
                                            'Followers',
-                                           domain=lambda self: [('res_model',
-                                                                 '=',
-                                                                 self._name
-                                                                 )])
+                                           domain=lambda self: [
+                                            ('res_model', '=',self._name)])
+
+    @api.model
+    def create(self, vals):
+        '''Inherited the create method to assign the roll no and std'''
+        if vals.get('student_id'):
+            student = self.env['student.student'].browse(
+                                            vals.get('student_id'))
+            vals.update({'roll_no_id': student.roll_no,
+                         'standard_id': student.standard_id.id
+                         })
+        return super(ExamResult, self).create(vals)
+
+    def write(self, vals):
+        '''Inherited the write method to update the roll no and std'''
+        if vals.get('student_id'):
+            student = self.env['student.student'].browse(
+                                            vals.get('student_id'))
+            vals.update({'roll_no_id': student.roll_no,
+                         'standard_id': student.standard_id.id
+                         })
+        return super(ExamResult, self).write(vals)
+
+    def unlink(self):
+        '''Inherited the unlink method to check the state at the deletion.'''
+        for rec in self:
+            if rec.state != 'draft':
+                raise ValidationError(_('''
+                    You can delete record in unconfirm state only!'''))
+        return super(ExamResult, self).unlink()
+
+    @api.onchange('student_id')
+    def onchange_student(self):
+        '''Method to get standard and roll no of student selected'''
+        self.standard_id = False
+        self.roll_no_id = False
+        if self.student_id:
+            self.standard_id = self.student_id.standard_id.id
```

Comment:
self.standard_id = self.student_id.standard_id.id or False will not require to define self.standard_id = False